RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducerFromSoA.cc

```diff
@@ -160,16 +160,23 @@ void PixelTrackProducerFromSoA::produce(edm::StreamID streamID,
 
   int32_t nt = 0;
 
-  for (int32_t it = 0; it < nTracks; ++it) {
+  //sort index by pt
+  std::vector<int32_t> sortIdxs(nTracks);
+  std::iota(sortIdxs.begin(), sortIdxs.end(), 0);
+  std::sort(
```

Comment:
tracks with quality < dup are not fitted: pt in such cases is not set (can be random, NaN as well)
I would suggest to first perform a partition to sent all track with q < minQuality_ to the back and then sort
the front part ( q>=minQuality_).

Reply:
> pt in such cases is not set (can be random, NaN as well)

Can they be set to some well-defined value ?
`-1.` ? `0.` ? `NaN` ?

I agree with the partition, though.

Reply:
@VinInn PixelTrackProducerFromSoA does not save the pixel tracks with (q < minQuality_),  
(https://github.com/cms-sw/cmssw/pull/38065/files#diff-88f203fe053ea5354390093fbb87f1670e7f7cfb2679b87b864381ad9514404cR175-R176)

In the end the output should be identical, even if I change sortIdxs (ie. we save only tracks with q > minQuality_ , sorted by pT).
Am I missing something?